fh-messaging/lib/fh_metrics_man.js

```diff
@@ -202,19 +202,65 @@ MetricsManager.prototype.deleteMetricsData = function(topic, date, callback) {
   messageMan = new fhmsg.Messaging(msgConfig, self.mLogger);
   self.messageMans.push(messageMan);
   messageMan.database.on('tearUp', function() {
-    messageMan.database.db.dropCollection(collectionName, callback);
-    //remove all the data that is older than the x number of days from the intermediate tables
-    async.each(['transactionsdesttemp', 'transactionsgeotemp'], function(tempCol, cb) {
-      var daysToCalculate = self.mConfig.agenda.jobs.metrics_rollup_job.daysToKeep || 31;
-      var beforeDate = new Date(date.getFullYear(), date.getMonth(), date.getDate() - daysToCalculate);
-      var query = {"_id.ts":{$lt: beforeDate.getTime()}};
-      messageMan.database.db.collection(tempCol, function(err, collection) {
-        if (err) {
-          return cb(err);
+
+    // Pruning for fhact
+    if (topic === 'fhact') {
+      var collection_names = [];
+      var prune_collection_names = [];
+      var filtered_collection_names = [];
+
+      // Retrieve the list of collections from the database
+      messageMan.database.db.listCollections({}).toArray( function(err, items) {
+        // get the collection names
+        for ( var i=0; i < items.length; i++ ) {
+          collection_names.push(items[i].name);
+        }
+
+        // filter out the collection names which do not contain "fhact_"
+        for ( var j=0; j<collection_names.length; j++ ) {
+          if ( collection_names[j].includes("fhact_") ) {
+            self.mLogger.debug('Found fhact collection: ' + collection_names[j]);
+            filtered_collection_names.push( collection_names[j] );
+          }
+        }
+
+        // iterate over each and construct Date from the "fhact_YYYYMMDD" name
+        var daysToCalculate = self.mConfig.agenda.jobs.metrics_rollup_job.daysToKeep || 31;
+        var pruning_date = new Date().getTime() - ( daysToCalculate * 24 * 60 * 60 * 1000 );
+        for ( var k=0; k<filtered_collection_names.length; k++ ) {
```

Comment:
How do you feel about rolling up all 3 loops into one? It is definitely more efficient but since the array it is looping over is so small, i am not too worried.